src/Executor/DBHyphenationExecutor.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Executor;
+
+use App\Hyphenator\ArrayHyphenator;
+use App\Model\Rule;
+use App\Model\Word;
+use App\Processor\DBProcessor;
+use App\Repository\RuleRepository;
+use App\Repository\WordRepository;
+
+
+class DBHyphenationExecutor implements ExecutorInterface
+{
+    public function __construct(private readonly \PDO $connection, private readonly string $word)
+    {
+
+    }
+
+    public function execute(): void
+    {
+        $wordRepo = new WordRepository($this->connection);
+        $ruleRepo = new RuleRepository($this->connection);
+
+        $word = $wordRepo->findByWord($this->word);
+        if ($word !== null && $word->getHyphenated() !== null) {
+            $matchedRules = $ruleRepo->getRulesMatchingWord($word);
+            echo 'Hyphenated word: ' . $word->getHyphenated() . PHP_EOL;
```

Comment:
A DBHyphenationExecutor service is now doing an output which means it is doing two jobs. Service should have single responsibility. Is this is necesseary for user to see? If yes you could inject IOService then and use it here instead, if it is not use logger or just remove all together.